Source/CesiumRuntime/Private/CesiumGltfPrimitiveComponent.h

```diff
@@ -105,16 +106,118 @@ class UCesiumGltfPrimitiveComponent : public UStaticMeshComponent {
 
   std::optional<Cesium3DTilesSelection::BoundingVolume> boundingVolume;
 
-  /**
-   * Updates this component's transform from a new double-precision
-   * transformation from the Cesium world to the Unreal Engine world, as well as
-   * the current HighPrecisionNodeTransform.
-   *
-   * @param CesiumToUnrealTransform The new transformation.
-   */
-  void UpdateTransformFromCesium(const glm::dmat4& CesiumToUnrealTransform);
+  void BeginDestroyPrimitive();
+};
+
+template <typename UnrealType>
+void UpdateTransformFromCesium(
```

Comment:
This function and `BeginDestroyPrimitive` below should be moved to the .cpp file and the anonymous namespace. If they need to be called externally, they should be added to the `ICesiumTile` interface (see above), and the virtual methods can be implemented by calling these templatized functions.